src/torrent.cpp

```diff
@@ -2018,7 +2018,8 @@ namespace libtorrent
 
 		// in case file priorities were passed in via the add_torrent_params
 		// and also in the case of share mode, we need to update the priorities
-		if (std::find(m_file_priority.begin(), m_file_priority.end(), 0) != m_file_priority.end())
+		if (!m_file_priority.empty() && std::find(m_file_priority.begin()
```

Comment:
Do you think sub-expression `!m_file_priority.empty()` should persist?

Reply:
no, but I think it should be fixed in a separate patch. ideally directly against master

Reply:
OK, RC_1_1 can live with this.